lucene/core/src/test/org/apache/lucene/search/TestScorerUtil.java

```diff
@@ -93,4 +94,20 @@ public void testLikelyImpactsEnum() throws IOException {
       }
     }
   }
+
+  public void testMinRequiredScore() {
+    double maxRemainingScore = random().nextDouble();
+    float minCompetitiveScore = random().nextFloat();
+    int numScorers = random().nextInt(1, 10);
```

Comment:
Let's test with high numbers of clauses too?

```suggestion
    int numScorers = random().nextInt(1, 1000);
```